external/js/cothority/src/network/connection.ts

```diff
@@ -61,8 +61,10 @@ export class WebSocketConnection {
      */
     constructor(addr: string, service: string) {
         const url = new URL(addr, {});
-        if (typeof window !== "undefined" && window.isSecureContext === true) {
-            url.set("protocol", "wss");
+        if (typeof globalThis !== "undefined" && typeof globalThis.location !== "undefined") {
```

Comment:
when not running the tests, `globalThis` isn't defined, so, never going to `wss`, even on secure pages?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis

Reply:
I learn everyday with MDN.